metadata/src/main/java/edu/unc/lib/dl/util/URIUtil.java

```diff
@@ -52,14 +52,16 @@ public static String join(String base, String... segments) {
 			}
 		}
 
-		for (String segment : segments) {
-			if (segment.charAt(0) != '/' && pathBuilder.length() > 0) {
-				pathBuilder.append('/');
-			}
-			if (segment.charAt(segment.length() - 1) == '/') {
-				pathBuilder.append(segment.substring(0, segment.length() - 1));
-			} else {
-				pathBuilder.append(segment);
+		if (segments[0] != null) {
```

Comment:
You are correct. This was a bug I discovered that was leading to a NPE. The varargs construction takes the null value passed in and stores it in segments[0].

Just to help me understand this better, what are some cases you think would be covered by your coding suggestion versus what I have now?